### ISSUE TITLE
Bad annotation crashes the cvterm extraction

### ISSUE THREAD

**Issue by exaexa** [LCSB-BioCore/SBML.jl#248]
cc @anandijain  -- thanks for reporting!

## Minimal code example to reproduce the problem

Get the model from https://www.ebi.ac.uk/biomodels/BIOMD0000000648#Files , then read as:

```
using SBML
readSBML("BIOMD0000000648_url.xml")
```

## Expected behavior

The model contains an invalid CV term:
```xml
	<bqmodel:hasProperty>
	<rdf:Bag>
	<rdf:li rdf:resource="http://identifiers.org/mamo/MAMO_0000046"/>
	</rdf:Bag>
	</bqmodel:hasProperty>
```
This should get ignored or reported as unknown CV term.

## Actual behavior

`libsbml` reports this as a `bqmodel:` CV term, but then reading the cvterm name as string fails and we get:
```
ERROR: ArgumentError: cannot convert NULL to string
Stacktrace:
  [1] unsafe_string
    @ ./strings/string.jl:84 [inlined]
  [2] unsafe_string
    @ ./c.jl:193 [inlined]
  [3] get_cv_term(cvt::Ptr{Nothing})
    @ SBML ~/work/SBML.jl/src/readsbml.jl:120
  [4] get_cv_terms(x::Ptr{Nothing})
    @ SBML ~/work/SBML.jl/src/readsbml.jl:100
  [5] get_model(mdl::Ptr{Nothing})
    @ SBML ~/work/SBML.jl/src/readsbml.jl:738
  [6] _readSBML(symbol::Symbol, fn::String, sbml_conversion::SBML.var"#69#71", report_severities::Vector{String}, throw_severities::Vector{String})
    @ SBML ~/work/SBML.jl/src/readsbml.jl:207
  [7] readSBML(fn::String, sbml_conversion::Function; report_severities::Vector{String}, throw_severities::Vector{String})
    @ SBML ~/work/SBML.jl/src/readsbml.jl:245
  [8] readSBML(fn::String, sbml_conversion::Function)
    @ SBML ~/work/SBML.jl/src/readsbml.jl:238
  [9] readSBML(fn::String)
    @ SBML ~/work/SBML.jl/src/readsbml.jl:238
 [10] top-level scope
    @ REPL[2]:1
```

## Optional: Suggestions for fixing

`CVTerm_getQualifierType` reports `MODEL_QUALIFIER` because the IRI prefix matches, but then `CVTerm_getModelQualifierType` returns unknown. Might be better to check that.

The model can be fixed by using `bqbiol:` prefix instead of `bqmodel:`.

**Comment by exaexa** [comment]
OK I fixed thsi in #249, will be released ASAP.

It might be nice to get feedback from libsbml on how to get any useful info from the CVTerms in case this happens. Surprisingly the "bad" part of BIOMD0000000648 gets erased even from the annotation string, which aiui should normally contain the leftover unparsed annotation.

@giordano @skeating @fbergmann -- opinions welcome :]

**Comment by fbergmann** [comment]
The CVTerms / ModelHistory API is only valid for the SBML approved RDF subset. While it is safe to read from the API what it can, when you use it to write, any unsupported RDF will be lost. For general RDF parsing you'll have to use the XMLNode from `getAnnotation` directly, or parse the string in whatever way you want. 

hope that helps.
